tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Http/WebRequest/HttpWebRequest_EndGetResponse_Integration.cs

```diff
@@ -71,11 +71,11 @@ public class HttpWebRequest_EndGetResponse_Integration
 
                 try
                 {
-                    scope = ScopeFactory.CreateOutboundHttpScope(Tracer.Instance, request.Method, request.RequestUri, WebRequestCommon.IntegrationId, out var tags, traceId: existingSpanContext?.TraceId, spanId: existingSpanContext?.SpanId, startTime);
+                    scope = ScopeFactory.CreateOutboundHttpScope(Tracer.Instance, request.Method, request.RequestUri, WebRequestCommon.IntegrationId, out _, traceId: existingSpanContext?.TraceId, spanId: existingSpanContext?.SpanId, startTime);
 
                     if (scope is not null)
                     {
-                        if (setSamplingPriority)
+                        if (setSamplingPriority && existingSpanContext.SamplingPriority != null)
```

Comment:
Based on nullable annotations, `existingSpanContext` could be null?
```suggestion
                        if (setSamplingPriority && existingSpanContext?.SamplingPriority is not null)
```

Reply:
Good catch!